### ISSUE TITLE
New rule: warn about setters without getters etc.

### ISSUE THREAD

**Issue by mzgol** [eslint/eslint#1638]
jsHint has a warning: "Setter is defined without getter." (see https://github.com/jshint/jshint/blob/3c81989d0fda302238608044d6fc191e61f7b9fa/src/messages.js#L154)

Example code triggering the error:
```js
var val;
var o = {
    set a(value) {
        val = value;
    },
};
```

AFAIK ESLint doesn't have any rule covering that.

**Comment by nzakas** [comment]
Sounds like a good idea.

**Comment by gyandeeps** [comment]
what should be the name of the rule?

**Comment by nzakas** [comment]
I think getters without setters are far more common, so I didn't warn on that be default, but maybe as an option?

How about `accessor-pairs` as the rule name?

**Comment by gyandeeps** [comment]
If i understand correctly,
getter without setter - Default warn
setter without getter - As an option

Name sounds good.

**Comment by mzgol** [comment]
@nzakas SGTM

**Comment by nzakas** [comment]
@gyandeeps setter without getter is the default warning, getter without setter is the optional one

**Comment by gyandeeps** [comment]
Working on this.

**Comment by michaeltreat** [comment]
m
